### PR TITLE
change: [M3-8509] - Add Pendo documentation to our development guide

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝
This PR creates some initial documentation for use of Pendo on Cloud Manager. See the internal doc for more details about access.

## Changes  🔄
- Updated the https://linode.github.io/manager/tooling/analytics.html page of our developer docs
   - Page heading is now Analytics; Adobe and Pendo are subheaders
   - Section on Pendo has been added
- Updated a comment in the usePendo.js hook to better reflect the usage of the hook

## Preview 📷

Screenshot 2024-10-17 at 9 32 46 AM

### Verification steps
(How to verify changes)
- Read the diff or check out this PR locally and `yarn docs`, then go to http://localhost:5173/manager/tooling/analytics.html#pendo
- Confirm the documentation makes sense and confirm there are no typos.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: docs/tooling/analytics.md [hunk 1]
```diff
@@ -53,10 +56,33 @@ These are the form events we use:
...
+Cloud Manager uses [Pendo](https://www.pendo.io/pendo-for-your-customers/) to capture analytics, guide users, and improve the user experience. To view Pendo dashboards, Cloud Manager developers must follow internal processes to request access.
+
+### Set Up and Initialization
+
+Pendo is configured in [`usePendo.js`](https://github.com/linode/manager/blob/develop/packages/manager/src/hooks/usePendo.ts). This custom hook allows us to initialize the Pendo analytics script when the [App](https://github.com/linode/manager/blob/develop/packages/manager/src/App.tsx#L56) is mounted.
+
+Important notes:
+
+- Pendo is only loaded if a valid `PENDO_API_KEY` is configured as an environment variable. In our development, staging, and production environments, `PENDO_API_KEY` is available at build time. See **Locally Testing Page Views & Custom Events and/or Troubleshooting Pendo** for set up with local environments.
+- We load the Pendo agent from the CDN, rather than [self-hosting](https://support.pendo.io/hc/en-us/articles/360038969692-Self-hosting-the-Pendo-agent).
```
Comment (abailly-akamai) [inline docs/tooling/analytics.md:78]: @mjac0bs unrelated to the documentation itself, do you know if we can pin a version? Considering this integration is very loosely typed, I am wondering if there's a risk of potential breaking updates we wouldn't be aware of getting a bundle straight from the CDN

**Review by abailly-akamai — APPROVED** [review]
Doc updates look great, thx for the attention to details 🥇

**Review by bnussman-akamai — APPROVED** [review]

File: docs/tooling/analytics.md (hunk 1 above)
Comment (mjac0bs) [reply docs/tooling/analytics.md:78]: @abailly-akamai We do have the ability to manually deploy a specific version of the agent in the Pendo settings - the drawback being that we'd need to monitor their releases (which seem to be [weekly](https://developers.pendo.io/agent-sdk-updates/)) for important security updates or bug fixes, and manually update to a new version.

Screenshot 2024-10-21 at 9 38 03 AM

Comment (abailly-akamai) [reply docs/tooling/analytics.md:78]: I am probably over-worrying - but hosting CDN libs is also sometimes considered a risk itself, so wondering about that in the case of Infosec. Pendo is an approved vendor so I am 99% this isn't an issue.

Comment (mjac0bs) [reply docs/tooling/analytics.md:78]: For now, we haven't been told by InfoSec or ProdSec that we **can't** use the CDN. (If that changes, we can update the way we're selecting our agent to be more specific - either self-hosting or manually set.) This came up in discussion before and they _are_ interested in documenting risks and additional safety checks we can include to help mitigate risks, even if they are low.

**Comment by mjac0bs** [comment]
Updated the PR to add one additional bullet point about making use of `data-testid` attributes; everything else is the same and ready to merge.

**Pull request merged**